stats/stats/src/update/new_blocks.rs

```diff
@@ -0,0 +1,389 @@
+    QueryFilter, QuerySelect, Set, Statement,
+};
+
+use super::UpdateError;
+
+#[derive(FromQueryResult)]
+struct NewBlocksData {
+    day: NaiveDate,
+    count: i64,
+}
+
+#[derive(Debug, FromQueryResult)]
+struct ChartData {
+    id: i32,
+    date: NaiveDate,
+}
+
+#[derive(Debug, FromQueryResult)]
+struct ChartId {
+    id: i32,
+}
+
+#[derive(Default, Debug)]
+pub struct Updater {}
+
+#[async_trait]
+impl super::UpdaterTrait for Updater {
+    async fn update(
+        &self,
+        db: &DatabaseConnection,
+        blockscout: &DatabaseConnection,
+    ) -> Result<(), UpdateError> {
+        let name = "new_blocks_per_day";
+        let last_row = ChartData::find_by_statement(Statement::from_sql_and_values(
+            DbBackend::Postgres,
+            r#"
+            SELECT distinct on (charts.id) charts.id, data.date 
+                FROM "chart_data_int" "data"
+                INNER JOIN "charts"
+                    ON data.chart_id = charts.id
+                WHERE charts.name = $1
+                ORDER BY charts.id, data.id DESC;
+            "#,
+            vec![name.into()],
+        ))
+        .one(db)
+        .await?;
+        let (id, data) = match last_row {
+            Some(row) => {
+                let data = NewBlocksData::find_by_statement(Statement::from_sql_and_values(
+                    DbBackend::Postgres,
+                    r#"
+                    SELECT date(blocks.timestamp) as day, COUNT(*)
+                        FROM public.blocks
+                        WHERE date(blocks.timestamp) >= $1
+                        GROUP BY day;
+                    "#,
+                    vec![row.date.into()],
+                ))
+                .all(blockscout)
+                .await?;
+                (row.id, data)
+            }
+            None => {
+                let data = NewBlocksData::find_by_statement(Statement::from_string(
+                    DbBackend::Postgres,
+                    r#"
+                    SELECT date(blocks.timestamp) as day, COUNT(*)
+                        FROM public.blocks
+                        GROUP BY day;
+                    "#
+                    .into(),
+                ))
+                .all(blockscout)
+                .await?;
+                let id = charts::Entity::find()
+                    .column(charts::Column::Id)
+                    .filter(charts::Column::Name.eq(name))
+                    .into_model::<ChartId>()
+                    .one(db)
+                    .await?
+                    .ok_or_else(|| UpdateError::NotFound(name.into()))?;
+                (id.id, data)
+            }
+        };
+
+        let data = data.into_iter().map(|row| chart_data_int::ActiveModel {
+            id: Default::default(),
+            chart_id: Set(id),
+            date: Set(row.day),
+            value: Set(row.count as i64),
+            created_at: Default::default(),
```

Comment:
You can make it the following way:

```suggestion
            chart_id: Set(id),
            date: Set(row.day),
            value: Set(row.count as i64),
            ..Default::default()
```

Reply:
yeah, I know, but in this case I wanted to provide all fields explicitly, so if someone adds new field, for example, he would need to look at this line and check, that this field should/shouldn't be default

also imo it's more readable, that you can see which fields are default and which are set

for tests that shouldn't be a problem, but for main logic I think it's a good idea